src/util/strutil.c

```diff
@@ -142,3 +143,48 @@ void str_ExtendBuffer
 	*bufferLen += extensionLen;
 	*buf = rm_realloc(*buf, sizeof(char) * *bufferLen);
 }
+
+// Utility function to check if the string matches
+// the regex pattern between from and to indices.
+bool str_MatchRegex
+(
+	const char* regex,   // regex pattern to match with
+	const char* str,     // string to match
+	int from,	         // starting index
+	int to		         // ending index
+) {
```

Comment:
Why do we need `from` and `to` I guess this is more general
but I think at the time being we need an exact match, does the string `str` is producible from the `pattern`

Reply:
@swilly22 I think I'll wrap it with exact match

Reply:
Sounds good.
Can you please give an example where one will use `from` to `to`

Reply:
@swilly22 I think that you are right, this is very edgy, I'll change it, thanks!